liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java

```diff
@@ -551,10 +551,14 @@ public boolean recompileBuildFile(File buildFile, List<String> compileArtifactPa
                 }
 
                 // update classpath for dependencies changes
-                compileArtifactPaths.clear();
-                compileArtifactPaths.addAll(project.getCompileClasspathElements());
-                testArtifactPaths.clear();
-                testArtifactPaths.addAll(project.getTestClasspathElements());
+                if (compileArtifactPaths != null) {
+                    compileArtifactPaths.clear();
+                    compileArtifactPaths.addAll(project.getCompileClasspathElements());
+                }
+                if (testArtifactPaths != null) {
+                    testArtifactPaths.clear();
+                    testArtifactPaths.addAll(project.getTestClasspathElements());
+                }
```

Comment:
If `compileArtifactPaths` or `testArtifactPaths` were null but the updated `project.getCompileClasspathElements()` or `project.getTestClasspathElements()` are not null, we should ensure that the paths still get updated.  But we can't initialize `compileArtifactPaths` or `testArtifactPaths` in this method if they are null.  So I think the solution would be to change the part where those variables are first initialized https://github.com/openliberty/ci.maven/blob/main/liberty-maven-plugin/src/main/java/io/openliberty/tools/maven/server/DevMojo.java#L807 so that they are never null e.g. initialize new ArrayLists at line 807/808 and add those elements

Reply:
Updated to initialize the two lists before adding elements.